packages/ipylintotype/src/ipylintotype/formatter.py

```diff
@@ -0,0 +1,88 @@
+import IPython
+import traitlets
+from ipykernel.comm import Comm
+
+from ._version import __comm__
+
+
+class AnnotationFormatter(IPython.core.interactiveshell.InteractiveShell):
+    annotators = traitlets.List()
+
+    comm_name = traitlets.Unicode(default_value=__comm__)
+
+    current_comm = traitlets.Any()
+
+    @traitlets.default("annotators")
+    def _default_annotators(self):
+        annotators = []
+        try:
+            from .annotators.mypy_annotator import MyPy
+
+            annotators += [MyPy()]
+        except ImportError:
+            self.log.warn("mypy could not be imported")
+
+        try:
+            from .annotators.pylint_annotator import PyLint
+
+            annotators += [PyLint()]
+        except ImportError:
+            self.log.warn("pylint could not be imported")
+
+        return annotators
+
+    def init_user_ns(self):
+        ...
+
+    def __init__(self, *args, **kwargs):
+        if "parent" in kwargs:
+            kwargs = {**kwargs["parent"]._trait_values, **kwargs}
+        super().__init__(*args, **kwargs)
+        self.init_comm()
+
+    def init_comm(self):
+        self.close()
+        self.current_comm = Comm(target_name=self.comm_name)
+        self.current_comm.on_msg(self.on_msg)
+
```

Comment:
@tonyfast are you proposing
```py
def init_comm(self):
    ...
    __import__('atexit').register(AnnotationFormatter.close)
```

Reply:
Exactly it belongs in the init_comm closure.